packages/common-server/src/files.ts

```diff
@@ -86,33 +91,51 @@ export function globMatch(patterns: string[] | string, fname: string): boolean {
   return _.some(patterns, (pattern) => minimatch(fname, pattern));
 }
 
-export function getAllFiles(opts: getAllFilesOpts): Dirent[] | string[] {
-  const { root, withFileTypes } = _.defaults(opts, {
+export async function getAllFilesWithTypes(
+  opts: GetAllFilesOpts
+): Promise<RespV2<Dirent[]>> {
+  const { root } = _.defaults(opts, {
     exclude: [".git", "Icon\r", ".*"],
-    withFileTypes: false,
   });
-  const allFiles = fs.readdirSync(root, { withFileTypes: true });
-  return _.reject(
-    allFiles.map((dirent) => {
-      const { name: fname } = dirent;
-      // match exclusions
-      if (
-        _.some([dirent.isDirectory(), globMatch(opts.exclude || [], fname)])
-      ) {
-        return null;
-      }
-      // match inclusion
-      if (opts.include && !globMatch(opts.include, fname)) {
-        return null;
-      }
-      if (withFileTypes) {
-        return dirent;
-      } else {
-        return dirent.name;
-      }
-    }),
-    _.isNull
-  ) as Dirent[] | string[];
+  try {
+    const allFiles = await fs.readdir(root, { withFileTypes: true });
+    return {
+      data: allFiles
+        .map((dirent) => {
+          const { name: fname } = dirent;
+          // match exclusions
+          if (
+            _.some([dirent.isDirectory(), globMatch(opts.exclude || [], fname)])
+          ) {
+            return null;
+          }
+          // match inclusion
+          if (opts.include && !globMatch(opts.include, fname)) {
+            return null;
+          }
+          return dirent;
+        })
+        .filter(isNotNull),
+      error: null,
+    };
+  } catch (err) {
+    return {
+      error: new DendronError({
+        message: "Error when reading the vault",
+        payload: err,
+        // Marked as minor to avoid stopping initialization. Even if we can't read one vault, we might be able to read other vaults.
```

Comment:
Can you move this comment to the docstring of the method? That way it's easier for someone using it to know that this is the behavior.